sparselsh/lsh.py

```diff
@@ -209,12 +227,12 @@ def index(self, input_points, extra_data=None):
             this is a target/class-value of some type.
         """
 
-        assert sparse.issparse(input_points), "input_points needs to be sparse"
-        if input_points.shape[0] != 1:
-            assert (extra_data is None) or \
-                (extra_data is not None and
-                 input_points.shape[0] == len(extra_data)), \
-                "input_points dimension needs to match extra data dimension"
+        assert issparse(input_points), "input_points needs to be a sparse matrix"
+        assert input_points.shape[1] == self.input_dim, "input_points wrong 2nd dimension"
+        assert input_points.shape[0] == 1 or (input_points.shape[0] > 1 and \
+               (extra_data is None or (isinstance(extra_data, list) and \
+               len(extra_data) == input_points.shape[0]))), \
+               "input_points dimension needs to match extra data dimension"
```

Comment:
Revert this please. It's unreadable and unnecessarily restricts the input types.